### ISSUE TITLE
Collectstatic error on 3.9.0

### ISSUE THREAD

**Issue by sondrelg** [yourlabs/django-autocomplete-light#1281]
Hi!

Thanks for maintaining this project, and releasing the newest version. When upgrading, I'm running into some issues when running the `collectstatic` command in Django.

```shell
❯ python manage.py collectstatic --no-input --clear

You have requested to collect static files at the destination
location as specified in your settings:

    /Users/sondrelg/Documents/project/build/staticfiles

This will overwrite existing files!
Are you sure you want to do this?

Type 'yes' to continue, or 'no' to cancel: yes
Post-processing 'vendor/select2/docs/plugins/highlight/css/pojoaque.css' failed!

Traceback (most recent call last):
  File "/Users/sondrelg/Documents/project/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/sondrelg/.virtualenvs/project/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/Users/sondrelg/.virtualenvs/project/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/sondrelg/.virtualenvs/project/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/sondrelg/.virtualenvs/project/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/Users/sondrelg/.virtualenvs/project/lib/python3.9/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 187, in handle
    collected = self.collect()
  File "/Users/sondrelg/.virtualenvs/project/lib/python3.9/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 134, in collect
    raise processed
whitenoise.storage.MissingFileError: The file 'vendor/select2/docs/plugins/highlight/css/pojoaque.jpg' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0x128daf940>.

The CSS file 'vendor/select2/docs/plugins/highlight/css/pojoaque.css' references a file which could not be found:
  vendor/select2/docs/plugins/highlight/css/pojoaque.jpg

Please check the URL references in this CSS file, particularly any
relative paths which might be pointing to the wrong location.
```

Any idea what might have caused this?

EDIT: Just to confirm, I downgraded to 3.8.2 and the issue went away.

**Comment by MDziwny** [comment]
Same issue with python 3.10.1:

```Post-processing 'vendor/select2/docs/themes/learn2/css/font-awesome.min.css' failed!

Traceback (most recent call last):
  File "/code/proxea/manage.py", line 12, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.10/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 187, in handle
    collected = self.collect()
  File "/usr/local/lib/python3.10/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 134, in collect
    raise processed
whitenoise.storage.MissingFileError: The file 'vendor/select2/docs/themes/learn2/fonts/fontawesome-webfont.woff2' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0x7f80aad1d6f0>.

The CSS file 'vendor/select2/docs/themes/learn2/css/font-awesome.min.css' references a file which could not be found:
  vendor/select2/docs/themes/learn2/fonts/fontawesome-webfont.woff2

Please check the URL references in this CSS file, particularly any
relative paths which might be pointing to the wrong location.```

**Comment by jpic** [comment]
Thanks for your feedback!! Removed 3.9.0 release for now until I figure the
problem

**Comment by MDziwny** [comment]
Good to know, it seems we're both (@sondrelg  and I) using [whitenoise](https://github.com/evansd/whitenoise/blob/06328669d928dd1e12921d7f6517835bfb2c1125/docs/index.rst)

**Comment by sondrelg** [comment]
If you need help testing a fix, let me know @jpic 👏

**Comment by MDziwny** [comment]
It could be linked to: https://github.com/evansd/whitenoise/issues/58, seems to be an issue from Django: https://code.djangoproject.com/ticket/21080

**Comment by jpic** [comment]
A pull request would help for sure :joy:

**Comment by hugorodgerbrown** [comment]
It looks like a problem with the `select2` import. The whitenoise error is highlighting that the css file is referencing a `woff2` file that it can't find. If I look at the source locally:

/usr/lib/python3.10/dist-packages/dal_select2/static/vendor/select2/docs/themes/learn2/css/font-awesome.min.css
```css
/*!
 *  Font Awesome 4.7.0 by @davegandy - http://fontawesome.io - @fontawesome
 *  License - http://fontawesome.io/license (Font: SIL OFL 1.1, CSS: MIT License)
ppy */
@font-face {
  font-family: "FontAwesome";
  src: url("../fonts/fontawesome-webfont.eot?v=4.7.0");
  src: url("../fonts/fontawesome-webfont.eot?#iefix&v=4.7.0")
      format("embedded-opentype"),
    url("../fonts/fontawesome-webfont.woff2?v=4.7.0") format("woff2"),
    url("../fonts/fontawesome-webfont.woff?v=4.7.0") format("woff"),
    url("../fonts/fontawesome-webfont.ttf?v=4.7.0") format("truetype"),
    url("../fonts/fontawesome-webfont.svg?v=4.7.0#fontawesomeregular")
      format("svg");
  font-weight: normal;
  font-style: normal;
}
...
```

If I then take a look at the fonts directory;

```
$ ls -al /usr/lib/python3.10/dist-packages/dal_select2/static/vendor/select2/docs/themes/learn2/fonts/
total 868
drwxr-xr-x 2 root root   4096 Jan 30 22:18 .
drwxr-xr-x 7 root root   4096 Jan 30 22:18 ..
-rw-r--r-- 1 root root 165742 Jan 30 22:18 fontawesome-webfont.eot
-rw-r--r-- 1 root root 444379 Jan 30 22:18 fontawesome-webfont.svg
-rw-r--r-- 1 root root 165548 Jan 30 22:18 fontawesome-webfont.ttf
-rw-r--r-- 1 root root  98024 Jan 30 22:18 fontawesome-webfont.woff
```

There is no `woff2` file.

**Comment by hugorodgerbrown** [comment]
Also running whitenoise - Django4.0, Python3.10.

**Comment by blueyed** [comment]
JFI: we've seen this with the file 'vendor/select2/docs/plugins/highlight/css/school-book.png' (and whitenoise).

**Comment by hugorodgerbrown** [comment]
Looks like the latest version have pulled in the whole select2 source, rather than just `dist` folder.

<img width="1072" alt="Screenshot 2022-01-31 at 10 20 04" src="https://user-images.githubusercontent.com/200944/151776757-acefe405-9059-42c0-9dad-11f00dd51ef9.png">

**Comment by hugorodgerbrown** [comment]
There is nothing wrong with the build script that I can see - the `select2` code is copied by the `copy-deps` script in `package.json`, and it seems fine. I've run the latest npm locally and it gives the expected outcome. I suspect rerunning the release script would work.

**Comment by sondrelg** [comment]
In case it's helpful: you could use https://test.pypi.org here, to generate as many test-releases as you want, if you're worried about using pypi 👍  See [this](https://packaging.python.org/en/latest/guides/using-testpypi/#using-testpypi-with-twine) for info about how to use it.

**Comment by jpic** [comment]
Is it fixed for you with whitenoise in 3.9.1 release?

I believe I ran the release.sh script inside an old/unclean git wc which uploaded a bunch of files from the select2 ... select2 submodule was removed  in a82d1501

**Comment by sondrelg** [comment]
Confirmed resolved for me 👍 Thanks!
